src/services/fileStorage/proxy-service.js

```diff
@@ -328,9 +330,18 @@ const fileStorageService = {
 			update.$unset = { parent: '' };
 		}
 
-		return permissionPromise()
+		const updatePromise = permissionPromise()
+			.then(() => updateParentDirectories(_id))
 			.then(() => FileModel.update({ _id }, update).exec())
 			.catch((err) => new Forbidden(err));
+
+		updatePromise
+			.then(() => updateParentDirectories(_id))
+			.catch((error) => {
+				logger.error(error);
+			});
```

Comment:
I think here you would need to use the async handler as well

Reply:
The function `asyncErrorHandler` was only defined in `fileStorageService#create` and the `asyncErrorLog` from `errors/utils` isn't used anywhere in this file. Which error handler should I use in your opinion?